lmnet/lmnet/networks/classification/base.py

```diff
@@ -180,17 +180,21 @@ def metrics(self, softmax, labels):
                 softmax = tf.Print(softmax,
                                    [tf.shape(softmax), tf.argmax(softmax, 1)], message="softmax:", summarize=200)
 
-            argmax_labels = tf.argmax(labels, 1)
+            argmax_labels = tf.cast(tf.argmax(labels, 1), tf.int32)
+
+            _, top_predicted_indices = tf.nn.top_k(softmax, k=1)
             accuracy, accuracy_update = tf.metrics.mean(
-                tf.cast(tf.nn.in_top_k(softmax, argmax_labels, k=1), tf.float32)
+                tf.cast(tf.reduce_any(tf.equal(tf.transpose(top_predicted_indices), argmax_labels), 0), tf.float32)
```

Comment:
The `tf.transpose` looks strange for me.
* `argmax_labels` shape is [batch_size,] and 
* k=3's `top_predicted_indices` shape is [batch_size, 3]

to apply `tf.equal` with broadcasting, we need to change whether shape.

But In my view point, It is readable that the`batch_size` order of shape is always first.
So, then, `argmax_labels` shape change to [batch_size, 1]
 ```
argmax_labels = tf.cast(tf.argmax(labels, 1), tf.int32)
argmax_labels = tf.expand_dims(argmax_labels, 1)

....
....
....

tf.cast(tf.reduce_any(tf.equal(top_predicted_indices, argmax_labels), axis=1), tf.float32)
```
This is better for me.

Reply:
yes ! thank you for the tip, that's much nicer to read.